src/charm.py

```diff
@@ -507,6 +508,15 @@ def _has_blocked_status(self) -> bool:
         """Returns whether the unit is in a blocked state."""
         return isinstance(self.unit.status, BlockedStatus)
 
+    @property
+    def s3_integrator_relation_exists(self) -> bool:
+        """Returns whether a relation with the s3 integrator exists."""
+        return bool(self.model.get_relation(S3_INTEGRATOR_RELATION_NAME))
+
+    def is_unit_blocked(self) -> bool:
```

Comment:
I named it as `is_unit_blocked` because in the k8s charm, the "blocked" states are actually `waiting` and `restarting`. Although, I agree, "blocked" can be confused with the BlockedStatus. renamed to `is_unit_busy` in 4bd1ce83fdedb811cea687e885db3a3deb8fcd72